apps/client/src/features/modals/AppSettingsModal.jsx

```diff
@@ -49,6 +51,9 @@ export default function AppSettingsModal() {
       pinCode: data.pinCode,
       timeFormat: data.timeFormat,
     });
+    getLatestVersion().then((data) => {
+      setVersionData(data);
+    });
```

Comment:
What if we only fetch the data if the user requests it. This request goes online, and I would think that the user should consent to this by triggering the action themselves
As you know, many video networks are behind offline or near-offline networks

If you think that the we should fetch the data either way, add a new useEffect for it with an empty dependency array
If we do in useEffect, we also need to add a cleanup function

so code would look something like

```
useEffect(() => {

    // we use abort controller to be able to cancel request
    const controller = new AbortController()
    const fetchData = async () => {
        axios.get(url, {signal: controller} )
        // handle stuff here
    }

    // cleanup function tells the component to cancel the fetch if component unmounts
    return(() => {
        controller.abort()
    })

// empty dependency array means useEffect runs in mount / unmount, only once
}, [])
```